src/mobile/src/ui/views/wallet/History.js

```diff
@@ -320,64 +369,117 @@ class History extends Component {
             };
         });
 
-        return orderBy(formattedTransfers, 'time', ['desc']);
+        return { transactions: orderBy(formattedTransfers, 'time', ['desc']), totals };
     }
 
-    renderIOTATransactions() {
+    renderIOTATransactions(transactions) {
         const {
-            theme: { primary, body },
+            theme: { primary, body, input },
             t,
             isRefreshing,
         } = this.props;
-        const data = this.prepIOTATransactions();
-        const noTransactions = data.length === 0;
+        const { filter, search } = this.state;
+
+        const hasNoTransactions = search === '' && filter === 'All' && transactions.length === 0;
+        const hasNoFilteredTransactions = (filter !== 'All' || search !== '') && transactions.length === 0;
+
+        if (search === '!help') {
+            const searchOptions = {
+                XYZ: t('history:searchHelpText'),
+                '100': t('history:searchHelpValue'),
+                '100Mi': t('history:searchHelpUnits'),
+                '>100': t('history:searchHelpMore'),
+                '<100i': t('history:searchHelpLess'),
+            };
+            return (
+                <View style={[styles.searchHelpContainer, { backgroundColor: input.bg }]}>
+                    {map(keys(searchOptions), (item) => (
+                        <View style={styles.searchHelpRow}>
```

Comment:
```suggestion
                    {map(keys(searchOptions), (item, index) => (
                        <View style={styles.searchHelpRow} key={index}>
```
This warning appears when `!help` is typed:
![image](https://user-images.githubusercontent.com/19519564/78182094-8878e580-7433-11ea-8784-30b3f13eb455.png)

We might be able to just use the indexes as keys since we don't plan to change the order of the items (https://reactjs.org/docs/lists-and-keys.html#keys)